assert/assertions_test.go

```diff
@@ -2566,3 +2567,99 @@ func TestIsNil(t *testing.T) {
 		t.Fatal("fail")
 	}
 }
+
+func TestUnwrapAll(t *testing.T) {
+	testCases := []struct {
+		in  error
+		out []error
+	}{
+		{
+			in: fmt.Errorf("abc: %w", fmt.Errorf("def: %w", fmt.Errorf("ghi"))),
+			out: []error{
+				errors.New("abc: def: ghi"),
+				errors.New("def: ghi"),
+				errors.New("ghi"),
+			},
+		},
+		{
+			in: fmt.Errorf("abc: %w", syscall.ENOENT),
+			out: []error{
+				fmt.Errorf("abc: %w", syscall.ENOENT),
+				syscall.ENOENT,
```

Comment:
Could you use some other error NOT from the `syscall` package? That package is too implementation dependent. Maybe `io.EOF`?

Reply:
I wanted an error that wasn't just another `*errors.errorString`. I'll make a custom error instead, and update.